File: src/main/java/io/teammion/morefood/recipes/SmeltingRecipes.java
package io.teammion.morefood.recipes;

import io.teammion.morefood.Items;
import io.teammion.morefood.Registry;
import net.minecraft.item.ItemStack;

/**
 * Created by Stefan Wimmer <stefanwimmer128@gmail.com> on 05.06.16.
 */
public class SmeltingRecipes
{
    public static void register()
    {
        Registry.addSmelting(
                new ItemStack(Items.DYE, 1, 3),
                new ItemStack(Items.COCOA_BEAN_ROASTED)
        );
        
        Registry.addSmelting(
                Items.SCHNITZEL_RAW,
                new ItemStack(Items.SCHNITZEL)
        );
        
        Registry.addSmelting(
                Items.EGG,
                new ItemStack(Items.EGG_BOILED)
        );
        
        Registry.addSmelting(
                Items.APPLE_CHOCOLATE,
                new ItemStack(Items.APPLE_CHOCOLATE_COATED)
        );
        
        Registry.addSmelting(
                Items.FISH_STICK_RAW,
                new ItemStack(Items.FISH_STICK)
        );
        
        Registry.addSmelting(
                Items.STRAWBERRY_CHOCOLATE,
                new ItemStack(Items.STRAWBERRY_CHOCOLATE_COATED)
        );
        
        Registry.addSmelting(
                Items.FRENCH_FRIES_RAW,
                new ItemStack(Items.FRENCH_FRIES)
        );
    }
}

File: src/main/scala/io/teammion/morefood/recipes/SmeltingRecipes.scala
/*
 * Copyright (c) 2016, Team Mion
 *
 * Permission to use, copy, modify, and/or distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */

package io.teammion.morefood.recipes

import io.teammion.morefood.{Items, Registry}
import net.minecraft.item.ItemStack

/**
  * Smelting recipes (sorted alphabetically by output)
  *
  * @author Stefan Wimmer <stefanwimmer128@gmail.com>
  */
object SmeltingRecipes
{
    def register() : Unit =
    {
        Registry.addSmelting(
            Items.APPLE_CHOCOLATE,
            Items.APPLE_CHOCOLATE_COATED.stack
        )
        
        Registry.addSmelting(
            Items.BREAD_DOUGH,
            new ItemStack(Items.BREAD)
        )
        
        Registry.addSmelting(
            Items.EGG,
            Items.EGG_BOILED.stack
        )
        
        Registry.addSmelting(
            Items.FISH_STICK_RAW,
            Items.FISH_STICK.stack
        )
        
        Registry.addSmelting(
            Items.FRENCH_FRIES_RAW,
            Items.FRENCH_FRIES.stack
        )
        
        Registry.addSmelting(
            new ItemStack(Items.DYE, 1, 3),
            Items.COCOA_BEAN_ROASTED.stack
        )
        
        Registry.addSmelting(
            Items.SCHNITZEL_RAW,
            Items.SCHNITZEL.stack
        )
        
        Registry.addSmelting(
            Items.STRAWBERRY_CHOCOLATE,
            Items.STRAWBERRY_CHOCOLATE_COATED.stack
        )
    }
}
